src/Illuminate/Http/TransformableResponse.php

```diff
@@ -0,0 +1,333 @@
+<?php
+
+namespace Illuminate\Http;
+
+use Illuminate\Support\Arr;
+use Illuminate\Support\Str;
+use Illuminate\Validation\ValidationData;
+use Illuminate\Contracts\Support\Arrayable;
+
+abstract class TransformableResponse extends JsonResponse
+{
+    /**
+     * Constructor.
+     *
+     * @param  mixed  $data
+     * @param  int    $status
+     * @param  array  $headers
+     * @param  int    $options
+     */
+    public function __construct($data = null, $status = 200, $headers = [], $options = 0)
+    {
+        if (is_array($data)) {
+            $data = $this->transform($data);
+        } elseif ($data instanceof Arrayable) {
+            $data = $this->transform($data->toArray());
+        }
+
+        parent::__construct($data, $status, $headers, $options);
+    }
+
+    /**
+     * Transforms the response data.
+     *
+     * @param  array $data
+     * @return array
+     */
+    public function transform(array $data)
+    {
+        $rules = $this->resolveArrayVisibilityRules($data, $this->visibilityRules());
+
+        $data = $this->applyVisibilityRules($data, $rules);
+
+        return $data;
+    }
+
+    /**
+     * Resolve array rules generating a new rule for every item in the array
+     * spcified with the '*' symbol.
+     *
+     * Example: posts.*.comments.*.title => posts.0.comments.0.title
+     *
+     * @param  array $data
+     * @param  array $rules
+     * @return array
+     */
+    protected function resolveArrayVisibilityRules(array $data, array $rules)
+    {
+        return array_reduce(array_keys($rules), function ($parsedRules, $rule) use ($data, $rules) {
+            if (Str::contains($rule, '*')) {
+                $gatheredRules = array_keys(
+                    ValidationData::initializeAndGatherData($rule, $data)
+                );
+
+                return array_merge(
+                    $parsedRules,
+                    $this->sanitizeArrayGatheredRules($rule, $gatheredRules, $rules[$rule])
+                );
+            }
+
+            $parsedRules[$rule] = $rules[$rule];
+
+            return $parsedRules;
+        }, []);
+    }
+
+    /**
+     * Sanitize gathered array rules removing those that don't appear into
+     * orginal rules.
```

Comment:
It's something we apply (ish) to the short descriptions to stop them getting too long. One rule that is strict is that they cannot overflow onto the next line.